src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3923,7 +3923,12 @@ and genMemberDefn node =
         // In implicit constructor, attributes should come even before access qualifiers
         ifElse ats.IsEmpty sepNone (sepSpace +> genOnelinerAttributes ats)
         +> genCtor
-        +> optPre (!- " as ") sepNone so genIdent
+        +> fun ctx ->
```

Comment:
A simpler check would be to use `sepSpace` like `optPre (sepSpace +> !- "as ") sepNone so genIdent`. The same check is already encapsulated there.